### PR TITLE
ghc: bump 902 -> 921

### PR DESCRIPTION
**Pull request by smunix**

### CODE REVIEW

File: package.yaml [hunk 1]
```diff
@@ -23,10 +23,11 @@ dependencies:
   - vector <= 1.0
 
 default-extensions:
-  - OverloadedStrings
+  - DerivingStrategies
+  - FlexibleContexts
```
Comment (smunix) [inline package.yaml:27]: ghc 9.2.1 requires this

File: flake.nix [hunk 2]
```diff
@@ -6,67 +6,73 @@
...
-    let
-      souffle = with final;
-      callPackage (import ./nix/souffle.nix { pkgs = final; }) { };
-      haskellPackages = super.haskell.packages.ghc902.override {
-        overrides = hself: hsuper: {
-          souffle-haskell = with final.haskell.lib;
-          with hself;
-          (overrideCabal
-          (addBuildTools (callCabal2nix "souffle-haskell" ./. { }) [
+    with fu.lib;
```
Comment (smunix) [inline flake.nix:9]: @luc-tielen, most of the code changes below are caused by `nixfmt` formatter. If you don't mind, let's enforce its usage prior to merging changes on nix files. What do you think?

File: package.yaml (hunk 1 above)
Comment (luc-tielen) [reply package.yaml:27]: Interesting :thinking:. So before it was somehow implicitly activated, while now this needs to be made explicit?

File: flake.nix (hunk 2 above)
Comment (luc-tielen) [reply flake.nix:9]: Yes, good suggestion. :+1: Let's do that in another PR though.

File: package.yaml (hunk 1 above)
Comment (smunix) [reply package.yaml:27]: Not sure how it worked before, here's the error I saw:
```sh
[1 of 9] Compiling Language.Souffle.Analysis

lib/Language/Souffle/Analysis.hs:121:10: error:
    • Non type-variable argument
        in the constraint: Category (Analysis m)
      (Use FlexibleContexts to permit this)
    • In the instance declaration for ‘Arrow (Analysis m)’
    |
121 | instance (Monad m, Monoid (m ()), Category (Analysis m)) => Arrow (Analysis m) where
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment (luc-tielen) [reply package.yaml:27]: Interesting, looks like they fixed a small gap in `FlexibleContexts` then! Probably caused by the `m ()` that's part of the `Monoid` constraint.

Adding the constraint to package.yaml is fine :+1:.

**Pull request merged**